### PR TITLE
feat(openai): Add a temperature input, Add the possibility to use a custom OpenAI model

### LINKED ISSUE #1959: OpenAI Connector models are fixed and point to outdated versions
## Issue

The OpenAI connector allow only the choice of _GPT-3.5 Turbo_ or _GPT-4_. That leads to two issues:

1. Impossible to pin model versions for reliable behaviour. These two are aliases and OpenAI changes what model they point to over time.
2. Multiple alternative and often newer and more performant models are unavailable to the connector user.

## Solution

Change Model field in `connectors/connectors/openai/element-templates/openai-connector.json` from a dropdown to a text field to allow any model input.

### PR DESCRIPTION
**Pull request by johnBgood**
## Description

This PR updates our **openai** connector:
- New `temperature` field
  - with a validation, the value must be between 0 and 2 (floats are ok) according to their [documentation](https://platform.openai.com/docs/api-reference/chat/create#chat-create-temperature)
- New **Custom** entry in the models dropdown.
  - When **Custom** is selected a new TextField appears, any value can be typed there

https://github.com/camunda/connectors/assets/1797846/b6cd2db8-c22b-47f2-a33c-c9b30652ed5b

## Related issues

closes #1959

### CODE REVIEW

**Review by igpetrov — APPROVED** [review]
LGTM!

Few suggestions:
- In the `Assignee` field put yourself
- In the project, where you selected `Connectors` you also need to change status to `In review`

File: connectors/openai/element-templates/openai-connector.json [hunk 1]
```diff
@@ -170,11 +166,16 @@
...
       "type": "Dropdown",
       "value": "gpt-3.5-turbo",
       "choices": [
+        {
+          "name": "Custom",
+          "value": "custom"
+        },
         {
           "name": "GPT-3.5-turbo",
           "value": "gpt-3.5-turbo"
```
Comment (sbuettner) [inline connectors/openai/element-templates/openai-connector.json:181]: Would be great if we could add the other known supported ones as well https://platform.openai.com/docs/models/continuous-model-upgrades namely: `gpt-4` and `gpt-4-turbo-preview`

Comment (sbuettner) [reply connectors/openai/element-templates/openai-connector.json:181]: Another reference: https://platform.openai.com/docs/models/model-endpoint-compatibility

Comment (johnBgood) [reply connectors/openai/element-templates/openai-connector.json:181]: Done ✔️

**Review by sbuettner — APPROVED** [review]
LGTM and congrats to your first PR! 🥳

**Pull request merged**